src/Core/Domain/ShowcaseCard/CommandHandler/CloseShowcaseCardHandler.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\ShowcaseCard\CommandHandler;
+
+use PrestaShop\PrestaShop\Core\Domain\ShowcaseCard\ConfigurationMap;
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Domain\ShowcaseCard\Command\CloseShowcaseCardCommand;
+
+/**
+ * Saves the showcase card status to keep it closed
+ */
+class CloseShowcaseCardHandler implements CloseShowcaseCardHandlerInterface
+{
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    /**
+     * @var ConfigurationMap
+     */
+    private $configurationMap;
+
+    /**
+     * CloseShowcaseCardHandler constructor.
+     *
+     * @param ConfigurationInterface $configuration
+     * @param ConfigurationMap $configurationMap
+     */
+    public function __construct(ConfigurationInterface $configuration, ConfigurationMap $configurationMap)
+    {
+        $this->configuration = $configuration;
+        $this->configurationMap = $configurationMap;
+    }
+
+    /**
+     * @param CloseShowcaseCardCommand $command
+     */
+    public function handle(CloseShowcaseCardCommand $command)
+    {
+        $configurationName = $this->configurationMap->getConfigurationNameForClosedStatus($command->getShowcaseCard());
+        $this->configuration->set($configurationName, '1');
```

Comment:
Why not sending an integer as param?

Reply:
For consistency's sake. Since values are actually stored as strings in `ps_configuration`, it will come back as a string when reading it from the DB. 

If I set it as an int, then the return type for `$configuration->get()` would be different depending on whether it was read from memory or from DB.

Reply:
there is PR to fix it https://github.com/PrestaShop/PrestaShop/pull/9193, which also points out some other issues with current `Configuration` adapter. @eternoendless do you think it should be (if at all should be) part of 1.7.6?